CRM/Contribute/Form/ManagePremiums.php

```diff
@@ -35,8 +37,7 @@ public function getDefaultEntity() {
   public function setDefaultValues() {
     $defaults = parent::setDefaultValues();
     if ($this->_id) {
-      $params = ['id' => $this->_id];
-      CRM_Contribute_BAO_Product::retrieve($params, $tempDefaults);
+      $tempDefaults = Product::get(FALSE)->addWhere('id', '=', $this->_id)->execute()->first();
```

Comment:
@seamuslee001 I'm just looking at you putting check permissions to FALSE. This is one of those places where ideally we WOULD have permissions kick in - ie this seems to be the form to configure products & hence the ability to get & save products should be material. I can think of 2 reasons why you put FALSE here
1) habit
2) our permissions for product are too tight & it failed without bypassing them

Reply:
ok @eileenmcnaughton I have changed this to not override check permissions now and have updated the entity permissions so that product matches contribution which seems to be the most accurate thing here IMO